File: src/data/courses/dsa/stacks/stack-basic-applications/remove-outermost-parentheses/code/python/remove_outermost_parentheses.py
# ============================================================
# Remove Outermost Parentheses
# ============================================================

class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        result, depth = [], 0
        for c in s:
            if c == '(':
                if depth > 0: result.append(c)
                depth += 1
            else:
                depth -= 1
                if depth > 0: result.append(c)
        return ''.join(result)

File: src/data/courses/dsa/stacks/stack-basic-applications/remove-outermost-parentheses/code/java/RemoveOutermostParentheses.java
class RemoveOutermostParentheses {
    public String removeOuterParentheses(String s) {
        StringBuilder result = new StringBuilder();
        int depth = 0;
        
        for (char c : s.toCharArray()) {
            if (c == '(') {
                if (depth > 0) result.append(c); // Skip outermost opening
                depth++;
            } else {
                depth--;
                if (depth > 0) result.append(c); // Skip outermost closing
            }
        }
        
        return result.toString();
    }
}
